OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -68,16 +68,18 @@ void IDemolishable.Demolish(Actor self, Actor saboteur, int delay, BitSet<Damage
 
 		void ITick.Tick(Actor self)
 		{
-			if (IsTraitDisabled)
+			if (IsTraitDisabled || actions.Count == 0)
 				return;
 
+			IEnumerable<int> modifiers = null;
 			foreach (var a in actions)
 			{
 				if (a.Delay-- <= 0)
 				{
-					var modifiers = self.TraitsImplementing<IDamageModifier>()
-						.Concat(self.Owner.PlayerActor.TraitsImplementing<IDamageModifier>())
-						.Select(t => t.GetDamageModifier(self, null));
+					if (modifiers == null)
+						modifiers = self.TraitsImplementing<IDamageModifier>()
+							.Concat(self.Owner.PlayerActor.TraitsImplementing<IDamageModifier>())
+							.Select(t => t.GetDamageModifier(self, null));
```

Comment:
Unless I'm missing something here, we should be able to cache both `self.TraitsImplementing<IDamageModifier>()` and `self.Owner.PlayerActor.TraitsImplementing<IDamageModifier>()` (and thus the `Concat`) in `INotifyCreated.Created` as they don't change over the lifetime of this trait? And then just run the `Select` here.